### PR TITLE
Use plain typedoc for generating nodejs docs

### PR DESCRIPTION
**Pull request by julienp**
### Proposed changes

Fixes https://github.com/pulumi/docs/pull/10964

Our custom `tsdocgen` tool is heavily dependant on the typescript version used in [pulumi/pulumi](http://github.com/pulumi/pulumi/) and changes to the version cause issues.

Similar to how we switched Python docs to use plain Sphinx, this moves to using plain typedoc to generate HTML (instead of our current 2 step process of typedoc generating JSON and then tsdocgen producing HTML from the).

Interesting changes are in https://github.com/pulumi/docs/pull/10964/commits/48e75a05b3b4d437b1301d40584e86f0ff218317

The URL structure for modules changed, there are redirects in https://github.com/pulumi/docs/pull/10964/commits/8601ac2d6f127cf41b8e1c6a0a46ae4d30139f51

Docs for pulumi/pulumi http://www-production-pulumi-docs-origin-pr-10964-f74f2df6.s3-website.us-west-2.amazonaws.com/docs/reference/pkg/nodejs/pulumi/pulumi/

### Related issues (optional)

https://github.com/pulumi/pulumi-hugo/pull/4045

### CODE REVIEW

**Comment by julienp** [comment]
Somehow the link in the menu pointing to the SDK docs loses the href attribute and I have no idea why this is happening

Screenshot 2024-03-12 at 15 50 44

File: config/_default/menus.yml [hunk 1]
```diff
@@ -215,7 +215,7 @@ languages:
   - name: SDK docs
     identifier: sdk-docs-javascript
     parent: javascript
-    pageRef: /docs/reference/pkg/nodejs/pulumi/pulumi
+    pageRef: https://www.pulumi.com/docs/reference/pkg/nodejs/pulumi/pulumi/
```
Comment (cnunciato) [inline config/_default/menus.yml:218]:
```suggestion
    url: https://www.pulumi.com/docs/reference/pkg/nodejs/pulumi/pulumi/
```

**Review by cnunciato — COMMENTED** [review]
@julienp This looks great! Added a note to get the link to work correctly.

If I wanted to make any thematic changes, how would I do that? I don't think we necessarily need to block on it, but would be good to know where to look when the time comes.

File: config/_default/menus.yml (hunk 1 above)
Comment (julienp) [reply config/_default/menus.yml:218]: Thanks!

**Comment by julienp** [comment]
> @julienp This looks great! Added a note to get the link to work correctly.
> 
> If I wanted to make any thematic changes, how would I do that? I don't think we necessarily need to block on it, but would be good to know where to look when the time comes.

Thanks for the help on the link!

Regarding theming the docs I see two options, we can include a CSS file via [--customCss](https://typedoc.org/options/output/#customcss) when we [run typedoc](https://github.com/pulumi/docs/blob/3d73c3e55a464d0d2ae391d01eb6e9c34c823750/scripts/run_typedoc.sh#L60), or we could create a [custom theme](https://typedoc.org/guides/themes/) and include that

**Comment by cnunciato** [comment]
@julienp There are a bunch, yeah, and from various places (see [this GitHub code search](https://github.com/search?q=org%3Apulumi+%2Fdocs%2Freference%2Fpkg%2Fnodejs&type=code&p=1)), so if we could redirect client-side without too much effort, that'd be ideal. We did something similar here after splitting a giant docs page into several:

https://github.com/pulumi/pulumi-hugo/blob/master/themes/default/content/docs/concepts/resources/_index.md?plain=1#L16-L18

**Comment by julienp** [comment]
I added client side `#anchor` redirects in https://github.com/pulumi/docs/pull/10964/commits/2ae4f61e01eddef546ebab8c2e077b47301ba361

The map redirects was gathered with a hacked up version of tscdocsgen.

**Comment by cnunciato** [comment]
@julienp Almost there! Looks like we just need a few redirects:

* For children belonging to packages other than `pulumi`, like https://www.pulumi.com/docs/reference/pkg/nodejs/pulumi/awsx/classic/. (Links to `awsx` are good, but to `awsx/classic` 404, as this page now lives at `awsx/modules/classic.html`.)

* For the anchors on these packages, like https://www.pulumi.com/docs/reference/pkg/nodejs/pulumi/awsx/classic/apigateway/#APIArgs (This currently 404s because of above, but would ideally land me at `/docs/reference/pkg/nodejs/pulumi/awsx/interfaces/classic.apigateway.APIArgs.html`.)

The client-side stuff seems doable in `redirects.js. For the others, we can probably create a new text file for them here, which will generate proper 301s once this goes out: https://github.com/pulumi/docs/tree/master/scripts/redirects  

Both should work in a PR preview as well. I'll see if I can spend a little time today setting some of this up for ya.

**Comment by cnunciato** [comment]
@julienp On second thought, the traffic to these pages probably doesn't warrant doing that much more work here. Looking at the data, I think it'd be fine if we just made sure these redirected to the right place:

```
/docs/reference/pkg/nodejs/pulumi/awsx/classic/apigateway/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/ecs/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/ec2/index.html
/docs/reference/pkg/nodejs/pulumi/cloud/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/classic/ecs/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/ecr/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/classic/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/lb/index.html
/docs/reference/pkg/nodejs/pulumi/terraform/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/autoscaling/index.html
/docs/reference/pkg/nodejs/pulumi/policy/index.html
/docs/reference/pkg/nodejs/pulumi/terraform/state/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/classic/cloudwatch/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/types/input/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/classic/ec2/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/classic/lambda/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/classic/lb/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/classic/dynamodb/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/classic/rds/index.html
/docs/reference/pkg/nodejs/pulumi/kubernetesx/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/classic/ecr/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/classic/sqs/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/classic/cloudfront/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/classic/s3/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/classic/sns/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/classic/cognito/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/types/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/classic/efs/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/classic/acmpca/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/classic/ebs/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/classic/codebuild/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/classic/autoscaling/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/types/output/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/cloudtrail/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/types/enums/ec2/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/classic/cloudtrail/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/types/enums/ecr/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/apigateway/index.html
/docs/reference/pkg/nodejs/pulumi/awsx/classic/billing/index.html
``` 

I'll push a PR that adds those of these that still need handling in a redirects file, and then we should be good to go. We can add the client-side redirects in a follow up, but I honestly wouldn't block on it. There's enough good here to roll forward with now IMO!

**Comment by cnunciato** [comment]
@julienp All right, https://github.com/pulumi/docs/pull/10964/commits/5995d843b6fd343dc29fd5101cb03e0e822a3488 adds a redirects file for the pages listed above, and replaces the `redirect_to:` pages you added with entries in that file. (I also noticed there doesn't seem to be a perfect new home for https://www.pulumi.com/docs/reference/pkg/nodejs/pulumi/pulumi/runtime/closure/index.html, so I [chose what seemed like the best option](https://github.com/pulumi/docs/pull/10964/commits/5995d843b6fd343dc29fd5101cb03e0e822a3488#diff-676be6b6894ad8e48fcf5f6fa013c9d87625e771742069ff01e17d1101684516R44). Let me know if there's a better one; we can easily change these if we have to.

If the preview looks good, I'll go ahead and merge, and we can address the anchor links in a quick follow. Thanks for making this happen!

**Pull request merged**